src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -320,37 +367,112 @@ private String getSamlResponseForAws(String oktaSessionToken) throws IOException
+    private String getSamlResponseForAwsRefresh() throws IOException {
+        Document document = launchOktaRefresh();
+        Elements samlResponseInputElement = document.select("form input[name=SAMLResponse]");
+        if (samlResponseInputElement.isEmpty()) {
+            throw new RuntimeException("You do not have access to AWS through Okta. \nPlease contact your administrator.");
+        }
+        return samlResponseInputElement.attr("value");
+    }
 
     private Document launchOktaAwsApp(String oktaSessionToken) throws IOException {
 
         HttpGet httpget = new HttpGet(oktaAWSAppURL + "?onetimetoken=" + oktaSessionToken);
-        try (CloseableHttpClient httpClient = HttpClients.createSystem();
-             CloseableHttpResponse responseSAML = httpClient.execute(httpget)) {
-
-            if (responseSAML.getStatusLine().getStatusCode() >= 500) {
+        CookieStore cookieStore = new BasicCookieStore();
+        Properties loadedProperties = new Properties();
+
+
+        loadedProperties.load(new FileReader(getCookies().toFile()));
+        loadedProperties.entrySet().stream()
+                .map(entry -> {
+                    BasicClientCookie basicClientCookie = new BasicClientCookie(entry.getKey().toString(), entry.getValue().toString());
+                    basicClientCookie
+                            .setDomain(oktaOrg);
+                    return basicClientCookie;
+                })
+                .forEach(cookieStore::addCookie);
+
+        try (CloseableHttpClient httpClient = HttpClients.custom().setDefaultCookieStore(cookieStore).useSystemProperties().build();
+             CloseableHttpResponse oktaAwsAppResponse = httpClient.execute(httpget)) {
+            if (oktaAwsAppResponse.getStatusLine().getStatusCode() >= 500) {
+                throw new RuntimeException("Server error when loading Okta AWS App: "
+                        + oktaAwsAppResponse.getStatusLine().getStatusCode());
+            } else if (oktaAwsAppResponse.getStatusLine().getStatusCode() >= 400) {
+                throw new RuntimeException("Client error when loading Okta AWS App: "
+                        + oktaAwsAppResponse.getStatusLine().getStatusCode());
+            }
+            Properties properties = new Properties();
+            cookieStore.getCookies().stream()
+                    .collect(Collectors.toMap(Cookie::getName, Cookie::getValue))
+                    .forEach(properties::setProperty);
+            properties.store(new FileWriter(getCookies().toFile()), "");
+            return Jsoup.parse(
+                    oktaAwsAppResponse.getEntity().getContent(),
+                    StandardCharsets.UTF_8.name(),
+                    oktaAWSAppURL
+            );
+        }
+    }
+    private Document launchOktaRefresh() throws IOException {
+        CookieStore cookieStore = new BasicCookieStore();
+        Properties loadedProperties = new Properties();
+        HttpGet httpget = new HttpGet(oktaAWSAppURL);
```

Comment:
:art: Same method as launchOktaAwsApp() with the only difference being the query params. Could hand in the URL or the HttpGet to make this cleaner.